server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java

```diff
@@ -238,10 +238,16 @@ private SortedKeyValueIterator<Key,Value> createIterator() throws IOException {
   public void close(boolean sawErrors) {
 
     if (memIters != null) {
+      log.debug("Returning mem iterators");
       tablet.returnMemIterators(memIters);
       memIters = null;
-      tablet.returnFilesForScan(fileReservationId);
-      fileReservationId = -1;
+      log.debug("Returning files for scanning");
+      try {
+        tablet.returnFilesForScan(fileReservationId);
+        fileReservationId = -1;
+      } catch (IllegalArgumentException | IllegalStateException e) {
+        log.debug("Error Returning files for scanning {}", e.getMessage());
```

Comment:
Do we not want to still throw an exception and then add the remainder of the close method in a finally block?

Reply:
Also I see that the returning of the files for scan may also result in the MetadataTableUtil.removeScanFiles at the end of the method which could throw a host of other errors, mostly variants of IOException.

Reply:
> Do we not want to still throw an exception and then add the remainder of the close method in a finally block?

It would be good to still throw the exception somehow.  If these exceptions are an indication that the scan was not correct, then do not want the scan to continue.  For example maybe a scan erroneously returns empty data instead of an error because an exception was logged and not rethrown.

Wondering if removing from the set could be moved earlier in the method.

Reply:
This is a bit outdated now, but the latest commit logs the current stack trace and then re-throws the exception for handling.